source/base/data_out_base.cc

```diff
@@ -7689,6 +7689,10 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
 #else
 
   const int myrank = Utilities::MPI::this_mpi_process(comm);
+  int rank;
+  MPI_Comm_size(comm, &rank);
+  MPI_Offset offset;
+  std::uint64_t size_placeholder;
```

Comment:
Always move variable declarations as close as possible to usage. You don't need those here.